src/attached_probe.cpp

```diff
@@ -62,7 +63,7 @@ bpf_prog_type progtype(ProbeType t)
 
 
 AttachedProbe::AttachedProbe(Probe &probe, std::tuple<uint8_t *, uintptr_t> func)
-  : probe_(probe), func_(func)
+  : probe_(probe), func_(func), progfd_(-1)
```

Comment:
I think this initialisation would be better somewhere else - if we create a new constructor in the future we could forget to copy this across. Either setting it in the class definition in `attached_probe.h` with a comment, or just before we start the `for..attempt` loop below would work and be safer.